src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java

```diff
@@ -51,6 +51,9 @@
 public class DeployMojo
     extends AbstractDeployMojo
 {
+    private static final Pattern ALT_INVALID_REPO_SYNTAX_PATTERN = Pattern.compile( "(.+)::(.+)::(.+)::(.+)" );
```

Comment:
I do not fully understand the purpose of this. You can extend this ad absurdum with zillions of double-colons. We cannot avoid this, but is not wrong:
`funny-id::https:///server/some/url/sub::/...`. I would rather drop this and make both other formats non-greedy all `n-1` matching groups.

Reply:
Done